commit/merkleroot/observation.go

```diff
@@ -318,22 +291,155 @@ type Observer interface {
 	ObserveOffRampNextSeqNums(ctx context.Context) []plugintypes.SeqNumChain
 
 	// ObserveLatestOnRampSeqNums observes the latest OnRamp sequence numbers for each configured source chain.
-	// NOTE: Make sure that caller supports the provided destination chain.
-	ObserveLatestOnRampSeqNums(ctx context.Context, destChain cciptypes.ChainSelector) []plugintypes.SeqNumChain
+	// NOTE: Make sure that caller supports the destination chain.
+	ObserveLatestOnRampSeqNums(ctx context.Context) []plugintypes.SeqNumChain
 
 	// ObserveMerkleRoots computes and returns the merkle roots for the provided sequence number ranges.
 	// NOTE: Make sure that caller supports the provided chains.
 	ObserveMerkleRoots(ctx context.Context, ranges []plugintypes.ChainRange) []cciptypes.MerkleRootChain
 
 	// ObserveRMNRemoteCfg observes the RMN remote config for the given destination chain.
 	// Check implementation specific details to learn if external calls are made, if values are cached, etc...
-	// NOTE: Make sure that caller supports the provided destination chain.
-	ObserveRMNRemoteCfg(ctx context.Context, dstChain cciptypes.ChainSelector) rmntypes.RemoteConfig
+	// NOTE: Make sure that caller supports the destination chain.
+	ObserveRMNRemoteCfg(ctx context.Context) rmntypes.RemoteConfig
 
 	// ObserveFChain observes the FChain for each supported chain. Check implementation specific details to learn
 	// if external calls are made, if values are cached, etc...
 	// NOTE: You can assume that every oracle can call this method, since data are fetched from home chain.
 	ObserveFChain(ctx context.Context) map[cciptypes.ChainSelector]int
+
+	// Close closes the observer and releases any resources.
+	Close() error
+}
+
+// asyncObserver is an Observer implementation that fetches the data asynchronously.
+type asyncObserver struct {
+	lggr                logger.Logger
+	syncObserver        observerImpl
+	cf                  context.CancelFunc
```

Comment:
```suggestion
	cancelFunc                  context.CancelFunc
```

`cf` is not sufficiently descriptive on its own to know what it is w/out an editor (i.e reading code on GH)